odoo/addons/base/models/res_users.py

```diff
@@ -215,6 +215,26 @@ def write(self, vals):
             self.env['ir.model.access'].call_cache_clearing_methods()
         return super(Groups, self).write(vals)
 
+    def _ensure_xml_id(self):
+        """Return the groups external identifiers, creating the external identifier for groups missing one"""
+        result = self.get_external_id()
+        missings = {group_id: f'__custom__.group_{group_id}' for group_id, ext_id in result.items() if not ext_id}
+        if missings:
+            self.env['ir.model.data'].create(
```

Comment:
Will it not raise concurrency error this creation because of unique sql-constraints?